operator/pkg/lib/sidecars/pods/get_test.go

```diff
@@ -0,0 +1,684 @@
+package pods_test
+
+import (
+	"context"
+	"fmt"
+	"github.com/kyma-project/istio/operator/api/v1alpha1"
+	"github.com/kyma-project/istio/operator/pkg/lib/sidecars/pods"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"testing"
+	"time"
+
+	. "github.com/kyma-project/istio/operator/pkg/lib/sidecars/test/helpers"
+)
+
+func createClientSet(t *testing.T, objects ...client.Object) client.Client {
+	err := v1alpha1.AddToScheme(scheme.Scheme)
+	require.NoError(t, err)
+	err = v1.AddToScheme(scheme.Scheme)
+	require.NoError(t, err)
+
+	fakeClient := fake.NewClientBuilder().WithScheme(scheme.Scheme).WithObjects(objects...).Build()
+
+	return fakeClient
+}
+
+func TestGetPodsForCNIChange(t *testing.T) {
+	ctx := context.Background()
+	enabledNamespace := FixNamespaceWith("enabled", map[string]string{"istio-injection": "enabled"})
+	disabledNamespace := FixNamespaceWith("disabled", map[string]string{"istio-injection": "disabled"})
+
+	tests := []struct {
+		name          string
+		c             client.Client
+		expectedImage pods.SidecarImage
+		isCNIEnabled  bool
+		wantEmpty     bool
+		wantLen       int
+	}{
+		{
+			name: "should not get any pod without istio-init container when CNI is enabled",
+			c: createClientSet(t,
+				NewSidecarPodBuilder().SetName("application1").SetNamespace("enabled").
+					SetInitContainer("istio-validation").SetPodStatusPhase("Running").Build(),
+				NewSidecarPodBuilder().SetName("application2").SetNamespace("enabled").
+					SetInitContainer("istio-validation").SetPodStatusPhase("Terminating").Build(),
+				enabledNamespace,
+			),
+			expectedImage: pods.SidecarImage{Repository: "istio/proxyv2", Tag: "1.10.0"},
+			isCNIEnabled:  true,
+			wantEmpty:     true,
+			wantLen:       0,
+		},
+		{
+			name: "should not get pods in system namespaces when CNI is enabled",
+			c: createClientSet(t,
+				NewSidecarPodBuilder().SetName("application1").SetNamespace("kube-system").Build(),
+				NewSidecarPodBuilder().SetName("application2").SetNamespace("kube-public").Build(),
+				NewSidecarPodBuilder().SetName("application3").SetNamespace("istio-system").Build(),
+				NewSidecarPodBuilder().SetName("application4").SetNamespace("enabled").Build(),
+				enabledNamespace,
+			),
+			expectedImage: pods.SidecarImage{Repository: "istio/proxyv2", Tag: "1.10.0"},
+			isCNIEnabled:  true,
+			wantEmpty:     false,
+			wantLen:       1,
```

Comment:
I think it would be better to not only rely on the size of the returned result, but also verify that the correct result is returned, e.g. by asserting the namespace.

Reply:
Discussion outcome: We should add the additional assertion.

Reply:
Added asserting pod name.